scripts/ai/AIDriveStrategyChopperCourse.lua

```diff
@@ -0,0 +1,782 @@
+    self.unloaders = {
+        currentUnloader = CpTemporaryObject(nil),
+        nextUnloader = CpTemporaryObject(nil)
+    }
+
+    -- Bool to set for special logic when unloader is need to be snugged up against the chopper so we don't drive in the fruit or off the field when unloading
+    self.chaseMode = false
+    return self
+end
+
+function AIDriveStrategyChopperCourse:setAllStaticParameters()
+    --Old Code to add markers to Choppers which don't have AI markers
+    self:checkMarkers()
+    AIDriveStrategyChopperCourse.superClass().setAllStaticParameters(self)
+
+    -- Check if we are a sugarcane havester for special handling
+    self.isSugarCaneHarvester = self:getSugarCaneHarvester()
+
+    -- We need set this as a variable and update left/right side on turns in self:updatePipeOffset()
+    self:setPipeOffsetX()
+    -- This is used to increase offset z when being chased instead along side
+    self:setPipeOffsetZ()
+
+    self:debug('AIDriveStrategyChopperCourse set')
+
+end
+
+function AIDriveStrategyChopperCourse:initializeImplementControllers(vehicle)
+    AIDriveStrategyChopperCourse:superClass().initializeImplementControllers(self, vehicle)
+    -- Need access to chopper controller functions
+    local _
+    _, self.pipeController = self:addImplementController(vehicle, PipeController, Pipe, {}, nil)
+    self.combine, self.chopperController = self:addImplementController(vehicle, ChopperController, Combine, {}, nil)
+end
+
+--------------------------------------------------------------------------------------------------------------------------------------------------------------
+-- Main Loop
+---------------------------------------------------------------------------------------------------------------------------------------------------------------
+
+function AIDriveStrategyChopperCourse:update(dt)
+    AIDriveStrategyCombineCourse.update(self, dt)
+    -- Old Code need to make Choppers work
+    self:updateChopperFillType()
+end
+
+function AIDriveStrategyChopperCourse:getDriveData(dt, vX, vY, vZ)
+    self:handlePipe()
+
+    if self.temporaryHold:get() then
+        self:setMaxSpeed(0)
+    end
+    if self.state == self.states.WORKING then
+        -- Harvesting
+        self:checkRendezvous()
+        self:checkBlockingUnloader()
+
+        if self:isChopperWaitingForUnloader() then
+            self:debug('No trailer calling for an unloader')
+            self:stopForUnload(self.states.WAITING_FOR_UNLOAD_ON_FIELD, true)
+        end
+
+    elseif self.state == self.states.TURNING then
+        self:checkBlockingUnloader()
+    elseif self.state == self.states.WAITING_FOR_LOWER then
+        if self:isChopperWaitingForUnloader() then
+            self:debug('No trailer calling for an unloader')
+            self:stopForUnload(self.states.WAITING_FOR_UNLOAD_ON_FIELD, true)
+        end
+    elseif self.state == self.states.UNLOADING_ON_FIELD then
+        -- Unloading
+        self:driveUnloadOnField()

```

Comment:
driveUnloadOnField does not make sense for a chopper as it is always unloading. This state and self.unloadState should be eliminated.